compiler/test/stdlib/number.test.gr

```diff
@@ -300,17 +439,75 @@ assert 1 / Number.sign(0.0) == Float64.toNumber(Float64.infinity)
 // TODO(#693): Replace with -Infinity literal when it exists
 assert 1 / Number.sign(-0.0) == Number.neg(Float64.toNumber(Float64.infinity))
 
-// Number.sin
+// Number.sin - These test a range as we approximate sin
 assert Number.sin(0.0) == 0
 assert Number.sin(1) > 0.84147 && Number.sin(1) < 0.84148
 assert Number.sin(-1) < -0.84147 && Number.sin(-1) > -0.84148
 assert Number.sin(20) > 0.91294 && Number.sin(20) < 0.91295
 assert Number.isNaN(Number.sin(Number.infinity))
 assert Number.isNaN(Number.sin(Number.nan))
 
-// Number.cos
+// Number.cos - These test a range as we approximate sin
 assert Number.cos(1) > 0.5403 && Number.cos(1) < 0.5404
 assert Number.cos(-1) > 0.5403 && Number.cos(-1) < 0.5404
 assert Number.cos(15) < -0.7596 && Number.cos(15) > -0.7597
 assert Number.isNaN(Number.cos(Number.infinity))
 assert Number.isNaN(Number.cos(Number.nan))
+
+// Number.tan - These test a range as we approximate sin
+assert Number.tan(1) > 1.5574 && Number.tan(1) < 1.5575
+assert Number.tan(-1) > -1.5575 && Number.tan(-1) < -1.5574
+assert Number.tan(15) < -0.8559 && Number.tan(15) > -0.8560
+assert Number.tan(1/2) > 0.5463 && Number.tan(1/2) < 0.5464
+assert Number.isNaN(Number.tan(Number.infinity))
+assert Number.isNaN(Number.tan(Number.nan))
+
+// Number.gamma
+// Note: Currently gamma will overflow the memory when using a bigint as such there are no tests for this
+assert Number.gamma(10.5) < 1133278.3889488 &&
+  Number.gamma(10.5) > 1133278.3889487
+assert Number.gamma(10) == 362880
+assert Number.gamma(1) == 1
+assert Number.gamma(0.5) == 1.7724538509055159
+assert Number.gamma(-0.5) < -3.544907 && Number.gamma(-0.5) > -3.544909
+assert Number.gamma(-1) == Number.infinity
+assert Number.gamma(-10) == Number.infinity
+assert Number.gamma(0.2) > 4.59084 && Number.gamma(0.2) < 4.59085
+assert Number.gamma(1/5) > 4.59084 && Number.gamma(1/5) < 4.59085
+assert Number.gamma(1/2) == 1.7724538509055159
+assert Number.isNaN(Number.gamma(Number.infinity))
+assert Number.isNaN(Number.gamma(Number.nan))
+
+// Number.factorial
+assert Number.factorial(10) == 3628800
+assert Number.factorial(5) == 120
+assert Number.factorial(0) == 1
+assert Number.factorial(-5) == -120
+assert Number.factorial(-10) == -3628800
+assert Number.factorial(10.5) == 11899423.083962297
+assert Number.factorial(0.5) == 0.886226925452759
+assert Number.isNaN(Number.factorial(Number.infinity))
+assert Number.isNaN(Number.factorial(Number.nan))
+
+// Number.toRadian
```

Comment:
I see rationals as the output for tests but not as input. Can a rational be used as input?